core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aspect/AASModelAspectGenerator.java

```diff
@@ -0,0 +1,57 @@
+package org.eclipse.esmf.aspectmodel.aspect;
+
+import org.apache.jena.rdf.model.Model;
+import org.eclipse.digitaltwin.aas4j.v3.model.Environment;
+import org.eclipse.esmf.aspectmodel.resolver.services.SammAspectMetaModelResourceResolver;
+import org.eclipse.esmf.aspectmodel.resolver.services.VersionedModel;
+import org.eclipse.esmf.aspectmodel.serializer.PrettyPrinter;
+import org.eclipse.esmf.aspectmodel.serializer.RdfModelCreatorVisitor;
+import org.eclipse.esmf.aspectmodel.vocabulary.Namespace;
+import org.eclipse.esmf.metamodel.Aspect;
+
+import java.io.*;
+import java.util.function.Function;
+
+public class AASModelAspectGenerator {
+
+    /**
+     * Generates an AASX archive file for a given Aspect and writes it to a given OutputStream provided by <code>nameMapper<code/>
+     *
+     * @param environment the Aspect for which an AASX archive shall be generated
+     * @param nameMapper a Name Mapper implementation, which provides an OutputStream for a given filename
+     * @throws IOException in case the generation can not properly be executed
+     */
+    public void generateTtlFile(final Environment environment, final Function<String, OutputStream> nameMapper )
+            throws IOException {
+        try ( final OutputStream output = nameMapper.apply( "env_name" ) ) {
+            output.write( generateAspectOutput( environment ).toByteArray() );
+        }
+    }
+
+    protected ByteArrayOutputStream generateAspectOutput(final Environment environment ) throws IOException {
+        final AASModelAspectTranslator visitor = new AASModelAspectTranslator();
+        final Aspect aspect = visitor.visitAas( environment ).get(0);
```

Comment:
Instead of `.get(0)`, the public interface of the class should provide a way to create an Aspect for each AAS submodel template (like AASModelAspectTranslator does, it returns a `List<>`).

Reply:
I did some updates but I think we have to understand how we will generate ttl file or ttl files for sobmodels.

Thanks,
Best Regards!